### PR TITLE
Get File from Uri

### LINKED ISSUE #397: Get file from Uri
Can we get File from android Uri.
Something like var file = uri.toFile()

### PR DESCRIPTION
**Pull request by Sathawale27**
Closes #397

### CODE REVIEW

**Comment by Sathawale27** [comment]
I signed it!

File: src/main/java/androidx/os/File.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import android.net.Uri
+import java.io.File
+
+/**
+ * Creates a File from the given Uri.
+ *
+ * @see File
+ */
+inline fun Uri.toFile(): File = File(path)
```
Comment (JakeWharton) [inline src/main/java/androidx/os/File.kt]: This should be in `Uri.kt` in `android/net/`

File: src/androidTest/java/androidx/os/FileTest.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package androidx.os
+
+import android.net.Uri
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.io.File
+
+class FileTest {
```
Comment (JakeWharton) [inline src/androidTest/java/androidx/os/FileTest.kt]: There's already `UriTest` where this should live.

File: src/main/java/androidx/os/File.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+package androidx.os
+
+import android.net.Uri
+import java.io.File
+
+/**
+ * Creates a File from the given Uri.
+ *
+ * @see File
```
Comment (JakeWharton) [inline src/main/java/androidx/os/File.kt]: Delete

File: src/main/java/androidx/os/File.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@file:Suppress("NOTHING_TO_INLINE") // Aliases to public API.
+
+package androidx.os
+
+import android.net.Uri
+import java.io.File
+
+/**
+ * Creates a File from the given Uri.
```
Comment (JakeWharton) [inline src/main/java/androidx/os/File.kt]: Please use braces `[ .. ]` around `File` and `Uri` as they are types.

**Comment by Sathawale27** [comment]
It returns the File so i thought it would be in File.kt. May i know why instrumented test is getting failed?
Thanks @JakeWharton.

**Comment by JakeWharton** [comment]
They're broken for external contributors with no easy fix in sight.

**Pull request merged**